builtin/providers/aws/resource_aws_nat_gateway.go

```diff
@@ -0,0 +1,169 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsNatGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNatGatewayCreate,
+		Read:   resourceAwsNatGatewayRead,
+		Delete: resourceAwsNatGatewayDelete,
+
+		Schema: map[string]*schema.Schema{
+			"allocation_id": &schema.Schema{
```

Comment:
Agreed about the concern - we default to the upstream API's terminology for a few reasons:

 * to ensure that users can read upstream AWS API docs and be clear about how Terraform is interacting: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateNatGateway.html 
 * to avoid the potential for re-thinking and discussion on every attribute name

For these reasons, the standard I've been using is to consider name changes at the Terraform level "only when there is significant UX benefit to Terraform users".

Applying that standard here, I believe it's going to be clear enough from documentation and context that it's best for us to stick with the upstream name.